### PR TITLE
feat: Move WAD server installation into a driver script

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
BREAKING CHANGE: WAD is does not get installed automatically anymore upon driver install

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,9 @@
     "platformNames": [
       "Windows"
     ],
+    "scripts": {
+      "install-wad": "./scripts/install-wad.mjs"
+    },
     "mainClass": "WindowsDriver"
   },
   "files": [
```
Comment (KazuCocoa) [inline package.json:44]: Need to add `scripts` and remove `install-npm.js`?

**Comment by eglitise** [comment]
Great idea! Although, it seems like this can only install the `.msi` versions now, so not the latest 1.2.99. I think that should at least be added to the readme.
Out of curiosity, what happens if WAD is already installed - does it just replace the current version with the new one?
Also I don't recall how the semantic release bot worked, but should the PR title have `feat!:` instead? Or would the breaking change comment be enough?

File: package.json (hunk 1 above)
Comment (mykola-mokhnach) [reply package.json:44]: good catch. I've almost  forgotten about it

**Comment by mykola-mokhnach** [comment]
> Also I don't recall how the semantic release bot worked, but should the PR title have feat!: instead? Or would the breaking change comment be enough?

It looks like the bot does not really understand exclamation marks in the title. It only pays attention to the presence of `BREAKING CHANGE:` magic in the commit message

**Comment by mykola-mokhnach** [comment]
> Although, it seems like this can only install the .msi versions now, so not the latest 1.2.99. I think that should at least be added to the readme.

Added support of 1.2.99

File: scripts/install-wad.mjs [hunk 2]
```diff
@@ -100,26 +112,55 @@ function selectRelease(releases, version) {
...
+function selectAsset(release) {
+  if (_.isEmpty(release.assets)) {
+    throw new Error(`WinAppDriver v${release.version} does not contain any matching releases`);
+  }
+  if (release.assets.length === 1) {
+    return release.assets[0];
+  }
+  // Since v 1.2.99 installers for multiple OS architectures are provided
+  for (const asset of release.assets) {
+    if (_.includes(asset.name, `win-${process.arch}.`)) {
```
Comment (eglitise) [inline scripts/install-wad.mjs:129]: The existing code has additional mapping for architectures (specifically, `process.arch` `x32` -> release `x86`). Not sure if that needs to be added here?

Comment (mykola-mokhnach) [reply scripts/install-wad.mjs:129]: I deliberately ignored x86 as node.js is only built for x64 arch

Comment (eglitise) [reply scripts/install-wad.mjs:129]: Isn't ia32 the same as 32-bit x86? Not actually sure what Node's `process.arch` returns on a 32-bit Windows.

Comment (mykola-mokhnach) [reply scripts/install-wad.mjs]: Added the mapping. I thought ia32 abbreviates Itanium rather than x86.

**Review by eglitise — APPROVED** [review]

**Review by KazuCocoa — APPROVED** [review]
lg

**Comment by KazuCocoa** [comment]
@Dor-bl FYI (if you're interested in)

**Pull request merged**